samples/cellular/modem_shell/src/sock/sock.c

```diff
@@ -505,6 +495,36 @@ static int sock_bind_to_pdp_context(struct sock_info *socket_info, int pdn_cid)
 	return err;
 }
 
+int sock_getaddrinfo(int family, int type, int flags, char *hostname, int pdn_cid)
+{
+	struct addrinfo *ai = NULL;
+	char addr_str[INET6_ADDRSTRLEN];
+	int err;
+
+	mosh_print("getaddrinfo query family=%d, type=%d, flags=%d, pdn_cid=%d, hostname=%s",
+		   family, type, flags, pdn_cid, hostname);
+
+	err = sock_getaddrinfo_req(&ai, family, type, flags, hostname, pdn_cid);
+	if (err) {
+		return err;
+	}
+
+	for (struct addrinfo *addr = ai; addr; addr = addr->ai_next) {
+		net_addr_ntop(addr->ai_family, &((struct sockaddr_in *)addr->ai_addr)->sin_addr,
+			      addr_str, sizeof(addr_str));
+		mosh_print("getaddrinfo answer family=%d, address=%s",
+			   addr->ai_family, addr_str);
+
+		if (addr->ai_canonname) {
```

Comment:
I tried the `-C` option, but did not get any canonical names. I checked the code and it seems that `nrf91_sockets.c` is missing implementation for copying the CNAME: https://github.com/nrfconnect/sdk-nrf/blob/f2d99e9c66e69d124eb051a7b3cef56a74b51fde/lib/nrf_modem_lib/nrf91_sockets.c#L308

I tried printing out the `ai_canonname` received from the modem, but it doesn't seem to work as expected. I checked from the modem log that the DNS query response contained a CNAME which was different from the requested hostname, but the response from the modem always contains the requested hostname. If the modem doesn't give the real CNAME, there's no point in fixing `nrf91_socket.c` and having the `-C` option here, either, I guess.

Reply:
Shall we create a task for this?

Reply:
I asked about this from the modem team and most likely this won't be changed for nRF91x1. If the `nrf91_socket.c` implementation would be completed, the canonical name would still be incorrect. Would it be better to leave the incorrect name out than provide it?